### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:

- Use common source url
- Mix format module config
- Add changelog to html doc
- Fix typos
- Set readme as default html doc page
- Badges and more badges!
- Set ex_doc module to the latest version

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -560,9 +561,9 @@ It's also necessary to create the DB table that will hold the feature flag data.
...
 
-The Redis PubSub adapter is the default and doesn't need to be excplicity configured. It can only be used in conjunction with the Redis persistence adapter however, and is not available when using Ecto for persistence. When used, it connects directly to the Redis instance used for persisting the flag data.
+The Redis PubSub adapter is the default and doesn't need to be explicitly configured. It can only be used in conjunction with the Redis persistence adapter however, and is not available when using Ecto for persistence. When used, it connects directly to the Redis instance used for persisting the flag data.
 
-The Phoenix PubSub adapter uses the high level API of `Phoenix.PubSub`, which means that under the hood it could use either its PG2 or Redis adapters, and this library doesn't need to know. It's provided as a convenient way to leverage distributed Erlang when using FunWithFlags in a Phoenix application, although it can be used independently (without the rest of the Phoenix framework) to add PubSub to Elixir apps running on Erlang clusters.
```
Comment (tompave) [inline README.md:565]: Trailing space is meaningful in Markdown. It adds a `<br>`.

File: README.md [hunk 2]
```diff
@@ -673,7 +674,7 @@ The Mixfile defines a few other helper tasks that allow to run the test suite wi
 
 ### Configuration changes have no effect in `MIX_ENV=dev`
 
-**Issue**: changing the library settings in the host application's Mix config file has no effect, or "missing process" exceptions are raised when booting. This should only be an issue in the development environment.
```
Comment (tompave) [inline README.md:676]: Here, too. Trailing space is meaningful in Markdown. It adds a <br>.

File: mix.exs [hunk 3]
```diff
@@ -244,19 +244,18 @@ defmodule FunWithFlags.Mixfile do
...
   defp docs do
     [
-      extras: ["README.md"],
-      main: "FunWithFlags",
-      source_url: "https://github.com/tompave/fun_with_flags/",
-      source_ref: "v#{@version}"
+      main: "readme",
+      source_url: @source_url,
+      source_ref: "v#{@version}",
+      extras: ["README.md", "CHANGELOG.md"]
```
Comment (tompave) [inline mix.exs:258]: I wouldn't change so many things in this file. I'd leave it as it is, and then maybe use the source URL and add the changelog here.

**Review by tompave — COMMENTED** [review]
Hi, thank you for the PR and taking care of the typos. I hope you just passed this though a spell checker rather than read everything line by line!

I'm happy to accept the typo fixes, but I can't accept the rest. Please see my comments.

I suggest you split the one commit this PR currently contains in multiple commits, so that you have one for the typos, one for a config change, etc.

**Comment by kianmeng** [comment]
> Hi, thank you for the PR and taking care of the typos. I hope you just passed this though a spell checker rather than read everything line by line!

:smile: 

> 
> I'm happy to accept the typo fixes, but I can't accept the rest. Please see my comments.
> 
> I suggest you split the one commit this PR currently contains in multiple commits, so that you have one for the typos, one for a config change, etc.

Thanks and appreciate for the feedback. I'll close this PR and opened two other PRs as suggested.

File: README.md (hunk 1 above)
Comment (kianmeng) [reply README.md:565]: > Trailing space is meaningful in Markdown. It adds a `<br>`.

TIL. :+1:

**Pull request closed without merging**